### PR TITLE
Add api diff job

### PR DESCRIPTION
**Pull request by acumino**
/kind enhancement

**What this PR does / why we need it**:
This PR adds a new test job for the gardener/gardener to calculate the incompatible api changes.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: config/jobs/gardener/gardener-apidiff-master.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    cluster: gardener-prow-build
+    skip_if_only_changed: "^docs/|\\.md$"
+    decorate: true
+    optional: true
+    branches:
+    - ^master$ # run on master branch
+    path_alias: gardener/gardener
+    spec:
+      containers:
+      - name: test-api-diff
```
Comment (timebertt) [inline config/jobs/gardener/gardener-apidiff-master.yaml:13]:
```suggestion
      - name: test
```

File: config/jobs/gardener/gardener-apidiff-master.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+presubmits:
+  gardener/gardener:
+  - name: pull-gardener-apidiff-master
+    cluster: gardener-prow-build
+    skip_if_only_changed: "^docs/|\\.md$"
+    decorate: true
+    optional: true
+    branches:
+    - ^master$ # run on master branch
+    path_alias: gardener/gardener
```
Comment (timebertt) [inline config/jobs/gardener/gardener-apidiff-master.yaml:10]: Shouldn't be needed as our modules are identified via default pattern `github.com/org/repo`.

File: config/jobs/gardener/gardener-apidiff-master.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+presubmits:
+  gardener/gardener:
+  - name: pull-gardener-apidiff-master
```
Comment (timebertt) [inline config/jobs/gardener/gardener-apidiff-master.yaml:3]:
```suggestion
  - name: pull-gardener-apidiff
```

and rename the file to `gardener-apidiff.yaml`

File: config/jobs/gardener/gardener-apidiff-master.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    skip_if_only_changed: "^docs/|\\.md$"
+    decorate: true
+    optional: true
+    branches:
+    - ^master$ # run on master branch
+    path_alias: gardener/gardener
+    spec:
+      containers:
+      - name: test-api-diff
+        image: golang:1.17
```
Comment (timebertt) [inline config/jobs/gardener/gardener-apidiff-master.yaml:14]:
```suggestion
        image: golang:1.17.7
```

**Review by timebertt — CHANGES_REQUESTED** [review]

**Comment by oliver-goetz** [comment]
/hold
until https://github.com/gardener/gardener/pull/5532 is merged
apart from Tim's feedback no more remarks

File: config/jobs/gardener/gardener-apidiff.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+presubmits:
+  gardener/gardener:
+  - name: pull-gardener-apidiff
+    cluster: gardener-prow-build
+    skip_if_only_changed: "^docs/|\\.md$"
+    decorate: true
+    optional: true
+    branches:
+    - ^master$ # run on master branch
```
Comment (timebertt) [inline config/jobs/gardener/gardener-apidiff.yaml:9]: Can you add the release branches?
```suggestion
    - ^master$
    - ^release-v\d+.\d+$
```

**Review by timebertt — COMMENTED** [review]
Nit

File: config/jobs/gardener/gardener-apidiff.yaml (hunk 5 above)
Comment (timebertt) [reply config/jobs/gardener/gardener-apidiff.yaml:9]: Right, I forgot about that.

**Review by timebertt — APPROVED** [review]
/lgtm

File: config/jobs/gardener/gardener-apidiff.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    spec:
+      containers:
+      - name: test
+        image: ghcr.io/gardener/ci-infra/golang-test:1.17.8
+        command:
+        - make
+        arge:
```
Comment (timebertt) [inline config/jobs/gardener/gardener-apidiff.yaml:16]: Noticed too late: 
```suggestion
        args:
```

**Pull request merged**